api/src/middleware/authorization-middleware.ts

```diff
@@ -0,0 +1,102 @@
+import { type NextFunction, type Response } from "express"
+import { type Request as JwtRequest } from "express-jwt"
+import { isNil } from "lodash"
+
+import logger from "@/utils/logger"
+import { User } from "@/models"
+
+import auth0Integration, { Auth0PayloadError } from "@/integrations/auth0-integration"
+
+export type AuthorizationRequest = JwtRequest & {
+  user?: User
+}
+
+const USER_CACHE_EXPIRATION_MS = 60 * 1000 // 1 minutes
+
+interface CachedUser {
+  promise: Promise<User>
+  timestamp: number
+}
+
+const userDebounceCache = new Map<string, CachedUser>()
+
+function cleanUpCache() {
+  const now = Date.now()
+  for (const [key, cached] of userDebounceCache.entries()) {
+    if (now - cached.timestamp > USER_CACHE_EXPIRATION_MS) {
+      userDebounceCache.delete(key)
+    }
+  }
+}
```

Comment:
It's always better to use a library than to role your own. I'd prefer if we used node-cache or some equivalent here.